src/Approximations/overapproximate.jl

```diff
@@ -516,3 +516,69 @@ function overapproximate(cap::Intersection{N,
     return overapproximate(swap(cap), dir; kwargs...)
 end
 
+"""
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle)  where {N}
+
+Decompose a lazy linear map of a cartesian product array.
+
+### Input
+
+- `lm`   -- lazy linear map of cartesian product array
+- `CartesianProductArray` -- type for dispatch
+- `oa` -- approximation option for decomposition
+
+### Output
+
+A `CartesianProductArray` representing the decomposed linear map.
+"""
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle) where {N}
+
+    @assert size(lm.M, 2) == dim(lm.X) "matrix needs to be commensurate with the cartesian product"
+
+    cpa = lm.X
+    M = lm.M
+
+    col_end_ind = 0
+
+    array = allocate_result(oa, N)
+    sizehint!(array,length(cpa.array))
+
+    i, n = 1, 0
```

Comment:
We use `n` for the dimension of a set. Just call them `row_start_ind` and `row_end_ind`?